test/NameRegistry.t.sol

```diff
@@ -2621,131 +2625,345 @@ contract NameRegistryTest is Test {
+        fnames[2] = "carol";
+        fnames[3] = "dan";
+
+        uint256[] memory tokenIds = new uint256[](4);
+        tokenIds[0] = ALICE_TOKEN_ID;
+        tokenIds[1] = BOB_TOKEN_ID;
+        tokenIds[2] = CAROL_TOKEN_ID;
+        tokenIds[3] = DAN_TOKEN_ID;
+
+        for (uint256 i = 0; i < fnames.length; i++) {
+            _register(users[i], fnames[i]);
+        }
 
-        _register(alice);
         uint256 biddableTs = block.timestamp + REGISTRATION_PERIOD + RENEWAL_PERIOD;
         _grant(MODERATOR_ROLE, ADMIN);
-        _requestRecovery(alice, recovery);
+
+        for (uint256 i = 0; i < fnames.length; i++) {
+            _requestRecovery(users[i], tokenIds[i], recovery);
+        }
 
         vm.warp(biddableTs);
-        vm.expectEmit(true, true, true, true);
-        emit Transfer(alice, POOL, ALICE_TOKEN_ID);
+        NameRegistry.ReclaimAction[] memory reclaimActions = new NameRegistry.ReclaimAction[](4);
+
+        for (uint256 i = 0; i < users.length; i++) {
+            reclaimActions[i] = NameRegistry.ReclaimAction(tokenIds[i], destinations[i]);
+            vm.expectEmit(true, true, true, true);
+            emit Transfer(users[i], destinations[i], tokenIds[i]);
+        }
         vm.prank(ADMIN);
-        nameRegistry.reclaim(ALICE_TOKEN_ID);
+        nameRegistry.reclaim(reclaimActions);
 
         // reclaim should extend the expiry ahead of the current timestamp
         uint256 expectedExpiryTs = block.timestamp + RENEWAL_PERIOD;
 
-        if (alice != POOL) assertEq(nameRegistry.balanceOf(alice), 0);
-        assertEq(nameRegistry.balanceOf(POOL), 1);
-        assertEq(nameRegistry.ownerOf(ALICE_TOKEN_ID), POOL);
-        assertEq(nameRegistry.expiryOf(ALICE_TOKEN_ID), expectedExpiryTs);
-        assertEq(nameRegistry.recoveryOf(ALICE_TOKEN_ID), address(0));
-        assertEq(nameRegistry.recoveryClockOf(ALICE_TOKEN_ID), 0);
+        for (uint256 i = 0; i < users.length; i++) {
+            assertEq(nameRegistry.balanceOf(users[i]), 0);
+            assertEq(nameRegistry.balanceOf(destinations[i]), 1);
+            assertEq(nameRegistry.expiryOf(tokenIds[i]), expectedExpiryTs);
+            assertEq(nameRegistry.ownerOf(tokenIds[i]), destinations[i]);
+            assertEq(nameRegistry.recoveryOf(tokenIds[i]), address(0));
+            assertEq(nameRegistry.recoveryClockOf(tokenIds[i]), 0);
+        }
     }
 
-    function testReclaimResetsERC721Approvals(address alice, address bob) public {
-        _assumeClean(alice);
-        _assumeClean(bob);
-        vm.assume(alice != bob);
+    function testReclaimResetsERC721Approvals(
+        address[4] calldata users,
+        address[4] calldata approveUsers,
+        address[4] calldata destinations
+    ) public {
+        address[] memory addresses = new address[](12);
+        for (uint256 i = 0; i < users.length; i++) {
+            _assumeClean(users[i]);
+            _assumeClean(approveUsers[i]);
+            _assumeClean(destinations[i]);
```

Comment:
can you add the _assumeClean call inside _assumeUnique and call it _assumeUniqueAndClean? would make these a lot easier to parse